### DISCUSSION TITLE
Errors on Neovim 0.10

### DISCUSSION THREAD

**Discussion by rogeriofrsouza** [LazyVim/LazyVim#3890] [Q&A] [answered]
I can't for the life of me run LazyVim in Neovim 0.10 😢 

I just installed the latest version for Linux [following the docs](https://github.com/neovim/neovim/blob/master/INSTALL.md#linux)
However, when I launch Neovim, some errors are thrown and I cannot run `:LazyHealth`, `:checkhealth` or debug anything.
Anyone experiencing this as well?

NVIM v0.10.0
Build type: Release

OS: Linux Mint 21.3 x86_64

![error1](https://github.com/LazyVim/LazyVim/assets/77423511/505c125b-5dd9-4b8d-b7d3-c052fd075b50)
![error2](https://github.com/LazyVim/LazyVim/assets/77423511/37e40a09-8fdc-475f-9541-d6dad1fb8bbf)
![error3](https://github.com/LazyVim/LazyVim/assets/77423511/18457df6-0382-4372-bce4-26c89c7d207a)
![error5](https://github.com/LazyVim/LazyVim/assets/77423511/e7c09154-4079-43c9-bd99-cf7a276289a5)

**Comment by folke** [comment]
What filesystem do you use?
This is an error triggered by Neovim's `vim.loader`.
Apparently your OS has a limit on file name length?

**Comment by dpetka2001** [comment]
I also use Linux Mint 21.3 and didn't observe any problem with a new LazyVim installation. I use default ext4 filesystem. Following screencast

[Screencast 2024-07-03 12:12:17.webm](https://github.com/LazyVim/LazyVim/assets/12776461/c78836a8-21c1-4c1a-b6a0-07d3be20e654)

**Reply by folke** [reply]
Yeah, ext4 or brtfs and others I know don't have this limit.
NTFS does however, but op seems to be using linux.

**Reply by rogeriofrsouza** [reply]
I also use the default ext4
![image](https://github.com/LazyVim/LazyVim/assets/77423511/da10c15d-cffd-4942-83d0-aebb59e00ade)

**Reply by rogeriofrsouza** [reply]
I think this problem in `vim.loader` is because `luac` cache file names gets too long on `ecryptfs`
And ecryptfs is enabled on my home directory 👀
 
https://github.com/neovim/neovim/issues/29372

**Reply by dpetka2001** [reply]
Do you have it encrypted?

**Reply by dpetka2001** [reply]
Yes just saw this on Neovim repo and that's why I asked. I don't use encryption.

**Reply by dpetka2001** [reply]
Can you try disabling the `vim.loader` in your `init.lua`? However, I'm not sure if maybe `lazy.nvim` enables it automatically by default maybe?

**Reply by dpetka2001** [reply]
Try in your `init.lua` to put `vim.loader.disable()` before `require("config.lazy")` and see if that makes a difference.

**Reply by folke** [reply]
you can also disable it from loading in lazy.nvim

**Reply by dpetka2001** [reply, marked as answer]
Just because i don't know this feature and want to make sure, is it `performance.cache.enabled` setting?

**Reply by folke** [reply]
yes, indeed! That gets enabled by default.
`vim.loader` is actually the lazy.nvim cache code that I upstreamed

**Reply by rogeriofrsouza** [reply]
Is it a good idea turning this off?
Is there a tradeoff?

**Reply by rogeriofrsouza** [reply]
Ok, it works! Thanks a lot guys 🥳
Tested with `performance.cache.enabled` set to false inside lazy setup in `lua/config/lazy.lua`

```lua
...
require("lazy").setup({
  spec = {
    -- add LazyVim and import its plugins
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    { import = "plugins" },
  },
  performance = {
    cache = {
      enabled = false,
    },
  },
})

```

**Reply by folke** [reply]
Neovim will be slower in loading lua files

**Reply by rogeriofrsouza** [reply]
Ok, I'll try to remove the encryption. Thanks!
